tests/ui/non_zero_suggestions.rs

```diff
@@ -0,0 +1,69 @@
+#![warn(clippy::non_zero_suggestions)]
+
+use std::num::{NonZeroI16, NonZeroI8, NonZeroU16, NonZeroU32, NonZeroU64, NonZeroU8, NonZeroUsize};
+
+fn main() {
+    // Positive test cases (lint should trigger)
+
+    // U32 -> U64
+    let x: u64 = 100;
+    let y = NonZeroU32::new(10).unwrap();
+    let r1 = x / u64::from(y.get());
```

Comment:
Oh, one more thing, try using [errors notations](https://rustc-dev-guide.rust-lang.org/tests/ui.html#error-annotations) on the test cases that "should trigger lint", so that it's much easier to identify the positive cases. So, in this case:

```rust
let r1 = x / u64::from(y.get());
//~^ ERROR: consider using `NonZeroU64::from()` for more efficient and type-safe conversion
```

Reply:
Done!

Reply:
Do I have to add notations for `HELP`? It just has `replace with` in all the messages. 😅

Reply:
If I remember correctly, `HELP` annotation doesn't work on suggestions, so just having `ERROR` should be fine.

And even if a suggestion turns out to be incorrect, ui-test could still catch them by testing the "fixed" code.